src/settings/SettingsModal.tsx

```diff
@@ -66,26 +69,79 @@ export const SettingsModal = (props: Props) => {
     useShowConnectionStats();
   const [enableE2EE, setEnableE2EE] = useEnableE2EE();
 
-  // Generate a `SelectInput` with a list of devices for a given device kind.
-  const generateDeviceSelection = (devices: MediaDevice, caption: string) => {
+  const lkRoom = useMaybeRoomContext();
+
+  const getTrackInUse = (sourceName: string): LocalTrack | undefined => {
+    if (!lkRoom) return undefined;
+
+    const t = Array.from(lkRoom.localParticipant.audioTracks.values()).find(
+      (d) => d.source === sourceName
+    );
+
+    return t?.track;
+  };
+
+  /**
+   * Generate a `SelectInput` with a list of devices for a given device kind.
+   * @param devices Info about the available & selected devices from MediaDevicesContext
+   * @param caption Device caption to be used if no labels available
+   * @param lkSource The livekit source name for this device type, eg. 'microphone': used to match up
+   *                 devices used by tracks in a current room (if any).
+   */
+  const generateDeviceSelection = (
+    devices: MediaDevice,
+    caption: string,
+    lkSource?: string
+  ) => {
     if (devices.available.length == 0) return null;
 
+    const trackUsedByRoom = lkSource ? getTrackInUse(lkSource) : undefined;
+
+    let selectedKey = devices.selectedId;
+    // We may present a different device as the currently selected one if we have an active track
+    // from the default device, because the default device may have changed since we acquired the
+    // track, in which case we want to display the one we're actually using rather than what the
+    // default is now.
+    if (
+      trackUsedByRoom &&
+      (devices.selectedId === "" ||
+        !devices.selectedId ||
+        devices.selectedId === "default")
+    ) {
+      // we work out what the actual device is based on groupId, but this only works if
+      // there is only one such device with the same group ID, which there won't be if
+      // we're using hardware with multiple sub-devices (eg. a multitrack soundcard)
+      const usedGroupId =
+        trackUsedByRoom?.mediaStreamTrack.getSettings().groupId;
+      const devicesWithMatchingGroupId = devices.available.filter(
+        (d) => d.groupId === usedGroupId && d.groupId !== "default"
```

Comment:
can the groupId ever be "default" ? Or doe we want to check for deviceId !== "default"

Reply:
Then I am quiet certain that you wanted to use deviceId in the comparison at the end of line 117 right?

Reply:
Ah, I see what you mean. I've changed that and also added a few more checks that we only do this in the right conditions.